functions/test_fact_difference_array.pp

```diff
@@ -0,0 +1,25 @@
+# @summary If the list of differences has any values fail the run and list the issues.
+#
+# If the list of differences has any values fail the run and list the issues.
+#
+# @example
+#   $fact_differences = $root_level_fact_differences + $classification_fact_differences
+#   classification::test_fact_difference_array($fact_differences)
+#
+# @param fact_differences An array of strings.
+#   These strings are generated by
+#   `classification::compare_root_level_facts` and
+#   `classification::compare_classification_facts`
+#
+# @return [Boolean]
+function classification::test_fact_difference_array(Array $fact_differences) >> Boolean {
+  $return = true
+
+  if $fact_differences.size() > 0 {
+    $return = false
+    fail((["Untrusted facts (left) don't match values from certname (right):"]
```

Comment:
This will never return here, so it will only ever return `true`, so there isn't really a reason to have a return at all.

Reply:
True. That was leftover from when I was trying to get `Deferred` working.

Reply:
Ah, cool